### DISCUSSION TITLE
Operating overloading and expr.Function

### DISCUSSION THREAD

**Discussion by purplefox** [antonmedv/expr#360] [General]
I have defined all my custom functions using expr.Function and added them as options during the compile step. This works nicely and my functions are then usable in expressions.

I would now like to overload some operators (e.g. "+", "-") to work on custom types. I followed the instructions for adding an Operator option. The custom function that does the adding is called "_add" and I have defined it using `expr.Function` just like all my custom functions.

Unfortunately this does not work, compile fails with:

```
panic: function _add for + operator does not exist in the environment [recovered]
	panic: function _add for + operator does not exist in the environment

goroutine 34 [running]:
testing.tRunner.func1.2({0x100890180, 0x1400010b760})
	/usr/local/go/src/testing/testing.go:1526 +0x1c8
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x364
panic({0x100890180, 0x1400010b760})
	/usr/local/go/src/runtime/panic.go:884 +0x1f4
github.com/antonmedv/expr/conf.(*Config).Check(0x1400017e1c0)
	/Users/tfox/go/pkg/mod/github.com/antonmedv/expr@v1.12.5/conf/config.go:85 +0x2a4
github.com/antonmedv/expr.Compile({0x1007dbac5, 0x7}, {0x1400019c0f0, 0x1e, 0x10088e440?})
	/Users/tfox/go/pkg/mod/github.com/antonmedv/expr@v1.12.5/expr.go:133 +0x184
```

I debugged into expr and this seems to be because the function "_add" does not exists in the `Types` map. It seems that only functions that exist in the env are added to this map, functions that are defined using `expr.Function` are not.

Is there a reason why operator overloading does not work with functions defined using `expr.Function`?

Thanks in advance!

**Comment by antonmedv** [comment]
This should be fixed

**Reply by purplefox** [reply]
Should I add an issue for this? :)

**Reply by antonmedv** [reply]
👍🏻
